kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ProxyChannelStateMachine.java

```diff
@@ -238,6 +252,9 @@ public void onClientUnwritable() {
     public void onClientWritable() {
         if (serverReadsBlocked) {
             serverReadsBlocked = false;
+            if (serverBackpressureTimer != null) {
+                serverBackpressureTimer.stop(proxyToServerBackpressureMeter);
+            }
```

Comment:
I look at this and think:

1. Perhaps we should `serverBackpressureTimer = null` after the call to stop?
2. If that's true, then do we need the separate `serverReadsBlocked` field at all? Couldn't we just use whether `serverBackpressureTimer == null`  to record the backpressuring state?

Reply:
1. Yes, we should.
2. Separation of concerns, yes we could get rid of the boolean but then we are conflating our metrics with our business logic. I think we should keep the two fields.